include/tvm/relax/nested_msg.h

```diff
@@ -275,6 +276,34 @@ NestedMsg<T> MapToNestedMsg(Expr expr, FType fmapleaf) {
   }
 }
 
+/*!
+ * \brief Map structinfo with possible nested-sinfo to nested message.
+ *
+ * This function will unpack recursive sinfo and run fmapleaf for each leaf,
+ * then recursively combines the results together into a NestedMsg.
+ *
+ * The nesting structure will corresponds to the tuple structure.
+ *
+ * \param sinfo The input struct info.
+ * \param fmapleaf The mapping function for each leaf with signature
+ *             NestedMsg<T> fmap(StructInfo)
+ * \tparam T the content type of nested msg
+ * \tparam FType The mapping function type
+ */
+template <typename T, typename FType>
+NestedMsg<T> MapToNestedMsg(StructInfo sinfo, FType fmapleaf) {
```

Comment:
We may need some tests for the new methods, in `tests/cpp/nested_msg_test.cc`.

Reply:
would be useful to isolate these two nested msg method out as a separate PR with unit-tests in tests/cpp/nested_msg_test.cc.